django/contrib/admindocs/utils.py

```diff
@@ -210,3 +211,21 @@ def replace_unnamed_groups(pattern):
         final_pattern += pattern[:start] + '<var>'
         prev_end = end
     return final_pattern + pattern[prev_end:]
+
+
+def replace_non_capturing_groups(pattern):
+    r"""
+    Finds non-capturing groups in the given `pattern` and removes them.
+    1. (?P<a>\w+)/b/(?:\w+)c(?:\w+) => /<a>/b/c
+    2. ^(?:\w+(?:\w+))a => /a
+    3. ^a(?:\w+)/b(?:\w+) => /a/b
+    """
+    group_start_end_indices = _find_groups(pattern, non_capturing_group_matcher)
+    final_pattern, prev_end = '', None
+    for start, end, _ in group_start_end_indices:
+        if prev_end:
+            final_pattern += pattern[prev_end:start]
+        else:
+            final_pattern += pattern[:start]
```

Comment:
`[None:start]` works the same as `[:start]`:
```suggestion
        final_pattern += pattern[prev_end:start]
```